### PR TITLE
[Feature](function) Support function assert_true

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

```sql
mysql> select assert_true(1, "wrong");
+------------------------------------------+
| assert_true(cast(1 as BOOLEAN), 'wrong') |
+------------------------------------------+
|                                        1 |
+------------------------------------------+
1 row in set (0.12 sec)

mysql> select assert_true(1, nullable("wrong"));
ERROR 1105 (HY000): errCode = 2, detailMessage = assert_true only accept constant for 2nd argument
mysql> select assert_true(0, "wrong");
ERROR 1105 (HY000): errCode = 2, detailMessage = (10.16.10.8)[INVALID_ARGUMENT][E33] wrong
```

doc pr: https://github.com/apache/doris-website/pull/1006

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/functions/function_assert_true.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        const auto& data = assert_cast<const ColumnBool*>(col_ptr.get())->get_data();
+
+        std::string errmsg =
+                assert_cast<const ColumnConst&>(*block.get_by_position(arguments[1]).column)
+                        .get_data_at(0)
+                        .to_string();
+
+        if (null_map) {
+            for (int i = 0; i < input_rows_count; i++) {
+                if (!data[i] || null_map->get_bool(i)) [[unlikely]] {
```
Comment (Mryange) [inline be/src/vec/functions/function_assert_true.cpp:85]: use ColumnNullable::has_null()

Comment (zclllyybb) [reply be/src/vec/functions/function_assert_true.cpp:85]: done

**Comment by zclllyybb** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Review by starocean999 — DISMISSED** [review]

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/AssertTrue.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * ScalarFunction 'assert_true'.
+ */
+public class AssertTrue extends ScalarFunction
+        implements UnaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/AssertTrue.java:39]:
```suggestion
        implements BinaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
```

**Comment by zclllyybb** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**